### PR TITLE
[BE] Upload activity cover image

### PR DESCRIPTION
**Pull request by MihaiDruta**
[BE] Upload activity cover image
https://trello.com/c/dMG8DN4K/82-be-upload-activity-cover-image

1.Added possibility to add a cover to activity
2.The cover is center cropped to a square and resized to all necessary dimensions (all dimensions are stored in services.yaml)
3.Added possibility to remove activity cover
4.Created specific documentation model for activity cover
5.Minor changes to functions related to User Avatar (DELETE request instead of POST for removing avatar, changed model name)

### CODE REVIEW

File: config/packages/nelmio_api_doc.yaml [hunk 1]
```diff
@@ -32,8 +32,8 @@ nelmio_api_doc:
       - { alias: ActivityTypeForActivityEdit, type: App\DTO\ActivityTypeDTO, groups: [ActivityEdit] }
   documentation:
     definitions:
-      Image:
-        description: Image
+      ImageUser:
+        description: UserAvatar
```
Comment (SpartakusMd) [inline config/packages/nelmio_api_doc.yaml:39]: Use a descriptive message to be clear about the usage.  Same for ImageActivity.

File: src/Controller/UserController.php [hunk 2]
```diff
@@ -528,8 +528,8 @@ public function getUserList(
 
     /**
      * Remove User avatar.
-     * @Rest\Post("/{id}/remove_avatar")
-     * * @SWG\Post(
+     * @Rest\Delete("/{id}/remove_avatar")
+     * * @SWG\Delete(
```
Comment (SpartakusMd) [inline src/Controller/UserController.php]: Remove a `*` symbol.

File: src/Service/ImageManager.php [hunk 3]
```diff
@@ -2,27 +2,165 @@
...
+    private $activityCoverSize;
 
-    public function __construct(UserAvatarManager $userAvatarManager)
+    private $currentTargetDirectory;
+    private $currentImageSize;
+
+    /**
+     * @param User|Activity $entity
+     */
+    private function initializeSizeDirectory($entity): void
```
Comment (SpartakusMd) [inline src/Service/ImageManager.php]: Something's fishy here :) we should discuss.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Service/ImageManager.php [hunk 4]
```diff
@@ -98,13 +60,13 @@ public function checkFileType(UploadedFile $file)
...
     /**
      * @param UploadedFile $uploadedFile
-     * @param User|Activity $targetEntity
+     * @param $filename
      */
-    public function saveImageInDirectory(UploadedFile $uploadedFile, $targetEntity): void
+    public function saveImageInDirectory(UploadedFile $uploadedFile, $filename): void
     {
-        $filename = $targetEntity->getId() . '.' . $uploadedFile->guessExtension();
+        //$filename = $targetEntity->getId() . '.' . $uploadedFile->guessExtension();
```
Comment (SpartakusMd) [inline src/Service/ImageManager.php]: Don't leave code comented. Remove it.

File: src/Service/ActivityCoverManager.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+class ActivityCoverManager extends ImageManager
+{
+    public function __construct(
+        $uploadDirectory,
+        ImageCropperResizer $cropperResizer,
+        $activityCoverDirectory,
+        $activityCoverSize
+    ) {
+        parent::__construct($uploadDirectory, $cropperResizer);
+        $this->initializeSizeDirectory($activityCoverDirectory, $activityCoverSize);
```
Comment (SpartakusMd) [inline src/Service/ActivityCoverManager.php]: You don't need this method. Do everything in the constructor from ImageManager.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Review by nspc97 — APPROVED** [review]

**Review by SpartakusMd — APPROVED** [review]

**Pull request merged**